error TS2552: Cannot find name 'VoidFunction'. Did you mean 'Function'?

**Describe the bug**

Suddenly started running into a typescript compilation error:

```
../../node_modules/@whatwg-node/promise-helpers/typings/index.d.ts(18,119): error TS2552: Cannot find name 'VoidFunction'. Did you mean 'Function'?
```

Based on this related issue in a separate repository, it seems that `VoidFunction` is not really recommended to use. 

https://github.com/lukeed/uvu/issues/89

**To Reproduce** Steps to reproduce the behavior:

Unsure

**Expected behavior**

No error

**Environment:**

- OS:
- `package-name...`:
- NodeJS:

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/.changeset/old-flowers-care.md
+++ b/.changeset/old-flowers-care.md
@@ -1,0 +1,5 @@
+---
+'@whatwg-node/promise-helpers': patch
+---
+
+Fix types by replacing `VoidFunction` type to `() => void`

--- a/packages/promise-helpers/src/index.ts
+++ b/packages/promise-helpers/src/index.ts
@@ -126,7 +126,7 @@ export function iterateAsync<TInput, TOutput>(
   iterable: Iterable<TInput>,
   callback: (
     input: TInput,
-    endEarly: VoidFunction,
+    endEarly: () => void,
     index: number,
   ) => MaybePromise<TOutput | undefined | null | void>,
   results?: TOutput[],

--- a/packages/promise-helpers/tests/handleMaybePromise.spec.ts
+++ b/packages/promise-helpers/tests/handleMaybePromise.spec.ts
@@ -161,7 +161,7 @@ describe('promise-helpers', () => {
       });
 
       describe('with sync function', () => {
-        let onFinally: jest.Mock<() => void)>;
+        let onFinally: jest.Mock<() => void>;
         let onError: jest.Mock;
         let onSuccess: jest.Mock;
 
@@ -228,7 +228,7 @@ describe('promise-helpers', () => {
       });
 
       describe('with fake promises', () => {
-        let onFinally: jest.Mock<() => void)>;
+        let onFinally: jest.Mock<() => void>;
         let onError: jest.Mock;
         let onSuccess: jest.Mock;
 

--- a/packages/promise-helpers/tests/handleMaybePromise.spec.ts
+++ b/packages/promise-helpers/tests/handleMaybePromise.spec.ts
@@ -161,7 +161,7 @@ describe('promise-helpers', () => {
       });
 
       describe('with sync function', () => {
-        let onFinally: jest.Mock<VoidFunction>;
+        let onFinally: jest.Mock<() => void)>;
         let onError: jest.Mock;
         let onSuccess: jest.Mock;
 
@@ -228,7 +228,7 @@ describe('promise-helpers', () => {
       });
 
       describe('with fake promises', () => {
-        let onFinally: jest.Mock<VoidFunction>;
+        let onFinally: jest.Mock<() => void)>;
         let onError: jest.Mock;
         let onSuccess: jest.Mock;
 

